nilearn/_utils/cache_mixin.py

```diff
@@ -145,6 +141,7 @@ def _safe_cache(memory, func, **kwargs):
         else:
             warnings.warn("Incompatible cache in %s: "
                           "old version of nibabel." % location)
+            return memory.cache(func, **kwargs)
```

Comment:
adding the return statement here will not prevent the versions JSON being overwritten with false information when `CHECK_CACHE_VERSION` is false. as this is a bit complex and independent of the simple type error in version comparisons I would suggest leaving that aside for now and maybe opening a separate issue.